dts/msm8974/msm8974pro-ab-pm8941-mtp.dts

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	// There are devices that identify themselves as msm8974pro-ab-pm8941-mtp,
+	// even when they are really not (at least physically).
+	// Therefore we need other things we can match on (e.g. the cmdline).
+	compatible = "qcom,msm8974-mtp", "qcom,msm8974", "qcom,mtp";
```

Comment:
Would be nice to drop the `"qcom,mtp"` here but that is a tiny nitpick that can be maybe fixed while merging.

Reply:
Those are the compatibles from downstream devicetree. Not sure what *should* be here with these "generic" devicetrees

Reply:
Just `"qcom,msm8974-mtp", "qcom,msm8974"` seems fine. The top level one is not really used.